### PR TITLE
Add image file size check

### PR DESCRIPTION
**Pull request by ItayZiv**
Fixes https://github.com/wpilibsuite/frc-docs/issues/693

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -9,6 +9,9 @@ BUILDDIR      = build
 LINTER        = doc8
 LINTEROPTS    = --ignore D001 # D001 is linelength
 LANGMAP       = es_MX: es, fr_CA: fr, he_IL: he, tr_TR: tr
+SIZECHECKER   = python scripts/imagesizechecker.py
+SIZEMAX       = 300
```
Comment (yellowjaguar5) [inline Makefile:13]:
```suggestion
SIZEMAX       = 500
```
Should this be `500`?

File: make.bat [hunk 2]
```diff
@@ -13,13 +13,17 @@ set SPHINXOPTS=-W --keep-going
 set LINTER=doc8
 set LINTEROPTS=--ignore D001 --ignore D002 --ignore D004
 set LANGMAP=es_MX: es, fr_CA: fr, he_IL: he, tr_TR: tr
+set SIZECHECKER=python scripts/imagesizechecker.py
+set SIZEMAX=300
```
Comment (yellowjaguar5) [inline make.bat:17]:
```suggestion
set SIZEMAX=500
```

File: Makefile [hunk 3]
```diff
@@ -9,6 +9,9 @@ BUILDDIR      = build
 LINTER        = doc8
 LINTEROPTS    = --ignore D001 # D001 is linelength
 LANGMAP       = es_MX: es, fr_CA: fr, he_IL: he, tr_TR: tr
+SIZECHECKER   = python scripts.imagesizechecker
+SIZEMAX       = 500
+
```
Comment (yellowjaguar5) [inline Makefile:14]:
```suggestion
EXCLUSIONSPATH= --exclude-file source/conf.py
```

File: Makefile [hunk 4]
```diff
@@ -9,6 +9,9 @@ BUILDDIR      = build
 LINTER        = doc8
 LINTEROPTS    = --ignore D001 # D001 is linelength
 LANGMAP       = es_MX: es, fr_CA: fr, he_IL: he, tr_TR: tr
+SIZECHECKER   = python scripts.imagesizechecker
```
Comment (yellowjaguar5) [inline Makefile:12]:
```suggestion
SIZECHECKER   = python3 -m scripts.imagesizechecker
```

**Comment by ItayZiv** [comment]
@Daltz333 this is probably ready for review and the check is not passing (LinkCheck seems to be failing because of artifactory)

**Comment by Daltz333** [comment]
Information about this needs to be added to the readme and the build guide

File: README.md [hunk 5]
```diff
@@ -46,6 +46,12 @@ Ensure the Python requirements are installed via running ``python3 -m pip instal
 
 This ensures that translations are updated concurrently by updating the transifex configuration file.
 
+### Image Size Check
+
+``.\make sizecheck`` or ``make sizecheck``
+
+This checks all image are under 500KB
```
Comment (Daltz333) [inline README.md:53]: Typo ```images``|

File: source/docs/contributing/frc-docs/build-instructions.rst [hunk 6]
```diff
@@ -78,6 +78,11 @@ Translate Check
 
 To ensure translations are updated concurrently, you must update the transifex configuration file. This can be done by running the ``.\make translate`` command. This **must** be done to to pass the buildbot.
 
+Image Size Check
+^^^^^^^^^^^^^^^^
+
+To makes sure all images which aren't excluded are under 500KB, run the size check by running ``.\make sizecheck``. This check will fail CI if it fails.
```
Comment (Daltz333) [inline source/docs/contributing/frc-docs/build-instructions.rst:84]: The language here seems odd. Perhaps change to 
```
Please run ``.\make sizecheck`` to verify that all images are below 500KB. This check *will* fail CI if it fails. Exclusions are allowed on a case by case basis and are added to the ``IMAGE_SIZE_EXCLUSIONS`` list in the configuration file.

**Pull request merged**